### PR TITLE
Server side pages and Admin UI

### PR DESCRIPTION
**Pull request by jerben**
This is a very early version and I wrote down many TODOs. I have a good idea now about the API that we need to properly support HTML pages next to JSON APIs.

### CODE REVIEW

File: sihl_core/lib/http.ml [hunk 1]
```diff
@@ -123,7 +193,17 @@ module Response = struct
...
     let code = Cohttp.Code.status_of_code resp.status in
     let body = resp.body |> Option.value ~default:Msg.ok_string in
-    Opium.Std.respond ~headers ~code (`String body)
+    let co_resp = Opium.Std.respond ~headers ~code (`String body) in
+    match resp.session with
+    | Nothing -> co_resp
+    | SetSession token ->
+        Opium.Std.Cookie.set ~http_only:true ~secure:false ~key:"session_id"
+          ~data:token co_resp
+    | StopSession ->
```
Comment (mikonieminen) [inline sihl_core/lib/http.ml:202]: StopSession sound weird, why not DeleteSession or EndSession?

File: sihl_users/lib/handler.ml [hunk 2]
```diff
@@ -172,119 +172,67 @@ end
...
 
-    let mycontent =
-      div
-        ~a:[ a_class [ "content" ] ]
-        [ h1 [ txt "A fabulous title" ]; txt "This is a fabulous content." ]
+    let handler =
+      Sihl_core.Http.get "/admin/dashboard/" @@ fun req ->
+      let user = Middleware.Authn.authenticate req in
+      Admin_ui.dashboard user |> Admin_ui.render |> Http.Response.html
+      |> Lwt.return
```
Comment (mikonieminen) [inline sihl_users/lib/handler.ml]: After looking at this and similar formatting, shouldn't this be either:
```
      Admin_ui.dashboard user |> Admin_ui.render |> Http.Response.html
        |> Lwt.return
```
or
```
      Admin_ui.dashboard user
      |> Admin_ui.render
      |> Http.Response.html
      |> Lwt.return
```

File: sihl_users/lib/handler.ml [hunk 3]
```diff
@@ -221,88 +221,65 @@ module AdminUi = struct
       |> Lwt.return
   end
 
+  module Users = struct
+    open Sihl_core
+
+    let handler =
+      Sihl_core.Http.get "/admin/users/users/" @@ fun req ->
+      let user = Middleware.Authn.authenticate req in
```
[the remark this answers is not in the document]
Comment (jerben) [reply sihl_users/lib/handler.ml:229]: Ocamlformat takes care of all code formatting, so I think yes.

File: sihl_users/lib/handler.ml (hunk 2 above)
Comment (jerben) [reply sihl_users/lib/handler.ml]: Ocamlformat takes care of all code formatting.

File: sihl_core/lib/http.ml (hunk 1 above)
Comment (jerben) [reply sihl_core/lib/http.ml]: Fixed.

**Pull request merged**